R/description.R

```diff
@@ -1,74 +1,89 @@
-#'     Version = "0.0.0.9000"
+#'     Language: es
 #'   )
 #' )
 #' ```
 #'
-#' @param fields A named list of fields to add to \file{DESCRIPTION},
-#'   potentially overriding the defaults. If `NULL`, retrieved from
-#'   `getOption("usethis.description")`, and (for backward compatibility) from
-#'   `getOption("devtools.desc")`.
+#' @param fields A named list of fields to add to DESCRIPTION, potentially
+#'   overriding default values. See [use_description()] for how you can set
+#'   personalized defaults using package options
 #' @export
 #' @examples
 #' \dontrun{
 #' use_description()
+#'
+#' use_description(fields = list(Language = "es"))
+#'
+#' use_description_defaults()
 #' }
 use_description <- function(fields = NULL) {
   name <- project_name()
   check_package_name(name)
+  if (!is.null(fields)) check_is_named_list(fields)
+  fields <- fields %||% list()
+  fields[["Package"]] <- name
 
-  fields <- fields %||%
-    getOption("usethis.description") %||%
-    getOption("devtools.desc") %||%
-    list()
-
-  desc <- build_description(name, fields)
+  desc <- build_description(fields)
   write_over(proj_get(), "DESCRIPTION", desc)
 }
 
-build_description <- function(name, fields = list()) {
-  desc_list <- build_description_list(name, fields)
+#' @rdname use_description
+#' @export
+use_description_defaults <- function() {
+  list(
+    usethis.description = getOption("usethis.description"),
+    devtools.desc = getOption("devtools.desc"),
+    usethis = list(
+      Package = "valid.package.name.goes.here",
+      Version = "0.0.0.9000",
+      Title = "What the Package Does (One Line, Title Case)",
+      Description = "What the package does (one paragraph).",
+      "Authors@R" = 'person("First", "Last", , "first.last@example.com", c("aut", "cre"))',
+      License = "What license it uses",
+      Encoding = "UTF-8",
+      LazyData = "true"
+    )
+  )
+}
+
+build_description <- function(fields = list()) {
+  desc_list <- build_description_list(fields)
 
   # Collapse all vector arguments to single strings
   desc <- vapply(desc_list, function(x) paste(x, collapse = ", "), character(1))
 
   paste0(names(desc), ": ", desc)
 }
 
-build_description_list <- function(name, fields = list()) {
-  author <- getOption("devtools.desc.author") %||%
-    'person("First", "Last", , "first.last@example.com", c("aut", "cre"))'
-  license <- getOption("devtools.desc.license") %||% "What license it uses"
-  suggests <- getOption("devtools.desc.suggests")
-
-  defaults <- list(
-    Package = name,
-    Version = "0.0.0.9000",
-    Title = "What the Package Does (One Line, Title Case)",
-    Description = "What the package does (one paragraph).",
-    "Authors@R" = author,
-    License = license,
-    Suggests = suggests,
-    Encoding = "UTF-8",
-    LazyData = "true",
-    ByteCompile = "true"
+build_description_list <- function(fields = list()) {
+  defaults <- use_description_defaults()
+  defaults <- utils::modifyList(
```

Comment:
It feels to me like this belongs in `use_description_defaults()`

Reply:
I actually made `use_description_defaults()` "data only" on purpose. It's the only easy way to see the usethis field defaults w/o reading source. Unless you care deeply, will leave this.